src/util/slab.rs

```diff
@@ -320,6 +337,51 @@ impl<T> Entry<T> {
     }
 }
 
+pub struct SlabIter<'a, T: 'a> {
+    slab: &'a Slab<T>,
+    cur_idx: isize,
+    yielded: isize
+}
+
+impl <'a, T> Iterator for SlabIter<'a, T> {
+    type Item = &'a T;
+
+    fn next(&mut self) -> Option<&'a T> {
+        while self.cur_idx < self.slab.cap && self.yielded < self.slab.len {
```

Comment:
Yeah, you're right. I originally just had the first condition and then added the second as an optimisation. I'll fix this soon.